packages/blockchain/algo/src/lib/services/algo.tx.ts

```diff
@@ -59,255 +85,264 @@ const prepareSignedTransaction = async (
-  const signedTxn = txn.signTxn(secretKey)
-
-  return Buffer.from(signedTxn).toString('hex')
+function isTransferAlgoKMS(
+  input: TransferAlgo | TransferAlgoKMS | ApiTransferAlgo | ApiTransferAlgoKMS,
+): input is TransferAlgoKMS {
+  return (input as TransferAlgoKMS).signatureId !== undefined
 }
 
-// TODO: OpenAPI bug -> missing from property
-export type TransferNftKMS = WithoutChain<ApiTransferNftKMS> & { from: string }
-export type TransferNft = WithoutChain<ApiTransferNft>
-
-const prepareTransferNFTSignedTransaction = async (
-  body: TransferNft | TransferNftKMS,
-  testnet = false,
-  algoWeb: AlgoWeb,
-  provider?: string,
-) => {
-  const algodClient = algoWeb.getClient(testnet, provider)
-  const params = await algodClient.getTransactionParams().do()
-
-  const decoder = new base32.Decoder({ type: 'rfc4648' })
-
-  const from = isWithSignatureId(body)
-    ? body.from
-    : algoWallet().generateAddressFromPrivatetKey(body.fromPrivateKey)
-
-  if (!body.value) {
-    throw new Error('No value specified')
+export const algoTxService = (args: { algoWeb: AlgoWeb }, apiCalls: AlgoApiCallsType) => {
+  const prepareSignedTransaction = async (
+    body: TransferAlgo | TransferAlgoKMS | ApiTransferAlgo | ApiTransferAlgoKMS,
+    testnet = false,
+    algoWeb: AlgoWeb,
+    provider?: string,
+  ): Promise<string> => {
+    const { balance } = await apiCalls.getBlockchainAccountBalance(body.account)
+    const requiredBalance = new BigNumber(body.amount).plus(body.fee || 0.001)
+    const accountBalance = new BigNumber(balance || 0)
+    if (accountBalance.isLessThan(requiredBalance)) {
+      throw new AlgoSdkError(
+        SdkErrorCode.INSUFFICIENT_FUNDS,
+        `Insufficient funds to create transaction from sender account ${
+          body.account
+        } -> available balance is ${accountBalance.toString()}, required balance is ${requiredBalance.toString()}.`,
+      )
+    }
+
+    return await prepareSignedTransaction(body, testnet, algoWeb, provider)
   }
 
-  const txn = algosdk.makeAssetTransferTxnWithSuggestedParams(
-    from,
-    body.to,
-    undefined,
-    undefined,
-    Number.parseInt(body.value),
-    undefined,
-    Number(body.contractAddress),
-    params,
-    undefined,
-  )
-
-  if (isWithSignatureId(body)) {
-    return JSON.stringify(txn)
+  const prepareCreateNFTSignedTransaction = async (
+    body: DeployNft | DeployNftKMS,
+    testnet = false,
+    algoWeb: AlgoWeb,
+    provider?: string,
+  ) => {
+    const algodClient = algoWeb.getClient(testnet, provider)
+    const params = await algodClient.getTransactionParams().do()
+
+    const decoder = new base32.Decoder({ type: 'rfc4648' })
+
+    const from = isWithSignatureId(body)
+      ? body.from
+      : algoWallet().generateAddressFromPrivatetKey(body.fromPrivateKey)
+
+    const txn = algosdk.makeAssetCreateTxnWithSuggestedParams(
+      from,
+      undefined,
+      1,
+      0,
+      false,
+      from,
+      undefined,
```

Comment:
![Screenshot 2022-10-27 at 20 16 42](https://user-images.githubusercontent.com/3228682/198367556-47247161-2c4c-4f9a-a982-6f3ba435b085.png)
you should use MintNftAlgorand DTO here with attr field, which could be included here instead of the undefined fields